go/internal/flag/flag_test.go

```diff
@@ -0,0 +1,308 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flag
+
+import (
+	goflag "flag"
+	"os"
+	"testing"
+
+	"github.com/spf13/pflag"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPreventGlogVFlagFromClobberingVersionFlagShorthand(t *testing.T) {
+	oldCommandLine := goflag.CommandLine
+	defer func() {
+		goflag.CommandLine = oldCommandLine
+	}()
+
```

Comment:
why do we need to mutate the global flagset to do these tests? from my reading we should be able to do our assertions with `testFlagSet` directly

Reply:
The function `PreventGlogVFlagFromClobberingVersionFlagShorthand` uses `f := goflag.Lookup("v")`, so we need to register `v` flag in `goflag.CommandLine` for this.